book/developers/exex/remote.md

```diff
@@ -1,3 +1,489 @@
 # Remote Execution Extensions
 
-WIP
+In this chapter, we will learn how to create an ExEx that emits all notifications to an external process.
+
+We will use [Tonic](https://github.com/hyperium/tonic) to create a gRPC server and a client.
+- The server binary will have the Reth client, our ExEx and the gRPC server.
+- The client binary will have the gRPC client that connects to the server.
+
+## Prerequisites
+
+See [section](https://github.com/hyperium/tonic?tab=readme-ov-file#dependencies) of the Tonic documentation
+to install the required dependencies.
+
+## Create a new project
+
+Let's create a new project. Don't forget to provide the `--lib` flag to `cargo new`,
+because we will have two custom binaries in this project that we will create manually.
+
+```console
+$ cargo new --lib exex-remote
+$ cd exex-remote
+```
+
+We will also need a bunch of dependencies. Some of them you know from the [Hello World](./hello-world.md) chapter,
+but some of specific to what we need now.
+
+```toml
+[package]
+name = "remote-exex"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+# reth
+reth = { git = "https://github.com/paradigmxyz/reth.git" }
+reth-exex = { git = "https://github.com/paradigmxyz/reth.git", features = ["serde"] }
+reth-node-ethereum = { git = "https://github.com/paradigmxyz/reth.git"}
+reth-tracing = { git = "https://github.com/paradigmxyz/reth.git" }
+
+# async
+tokio = { version = "1", features = ["full"] }
+tokio-stream = "0.1"
+futures-util = "0.3"
+
+# grpc
+tonic = "0.11"
+prost = "0.12"
+bincode = "1"
+
+# misc
+eyre = "0.6"
+
+[build-dependencies]
+tonic-build = "0.11"
+
+[[bin]]
+name = "exex"
+path = "src/exex.rs"
+
+[[bin]]
+name = "consumer"
+path = "src/consumer.rs"
+```
+
+As you can see, we also added a build dependency for Tonic. We will use it to generate the Rust code for our
+Protobuf definitions at compile time.
+
```

Comment:
can we link to the tonic docs here as well?

Reply:
Added a link to their hello world tutorial